veggieseasons/lib/main.dart

```diff
@@ -83,14 +94,141 @@ class _VeggieAppState extends State<VeggieApp> with RestorationMixin {
+                );
+              },
+              routes: [
+                GoRoute(
+                  path: '/list',
+                  pageBuilder: (context, state) {
+                    return FadeTransitionPage(
+                      restorationId: 'route.list',
+                      child: Builder(builder: (context) {
+                        return const ListScreen(restorationId: 'list');
+                      }),
+                    );
+                  },
+                  routes: [
+                    _buildDetailsRoute(),
+                  ],
+                ),
+                GoRoute(
+                  path: '/favorites',
+                  pageBuilder: (context, state) {
+                    return FadeTransitionPage(
+                      restorationId: 'route.favorites',
+                      child: Builder(builder: (context) {
+                        return const FavoritesScreen(
+                            restorationId: 'favorites');
+                      }),
+                    );
+                  },
+                  routes: [
+                    _buildDetailsRoute(),
+                  ],
+                ),
+                GoRoute(
+                  path: '/search',
+                  pageBuilder: (context, state) {
+                    return FadeTransitionPage(
+                      restorationId: 'route.search',
+                      child: Builder(builder: (context) {
+                        return const SearchScreen(restorationId: 'search');
+                      }),
+                    );
+                  },
+                  routes: [
+                    _buildDetailsRoute(),
+                  ],
+                ),
+                GoRoute(
+                  path: '/settings',
+                  pageBuilder: (context, state) {
+                    return FadeTransitionPage(
+                      restorationId: 'route.settings',
+                      child: Builder(builder: (context) {
+                        return const SettingsScreen(restorationId: 'settings');
+                      }),
+                    );
+                  },
+                  routes: [
+                    GoRoute(
+                      parentNavigatorKey: _rootNavigatorKey,
+                      path: 'categories',
+                      pageBuilder: (context, state) {
+                        return VeggieCategorySettingsScreen.pageBuilder(
+                            context);
+                      },
+                    ),
+                    GoRoute(
+                      parentNavigatorKey: _rootNavigatorKey,
+                      path: 'calories',
+                      pageBuilder: (context, state) {
+                        return CalorieSettingsScreen.pageBuilder(context);
+                      },
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ],
+        ),
       ),
     );
   }
+
+  GoRoute _buildDetailsRoute() {
```

Comment:
The details route is shared between the `/list`, `/favorites` and `/search` paths, but works exactly the same